test/server/config_validation/xds_fuzz.cc

```diff
@@ -0,0 +1,246 @@
+    sotw_or_delta_ = Grpc::SotwOrDelta::Sotw;
+  } else {
+    sotw_or_delta_ = Grpc::SotwOrDelta::Delta;
+  }
+}
+
+/**
+ * initialize an envoy configured with a fully dynamic bootstrap with ADS over gRPC
+ */
+void XdsFuzzTest::initialize() {
+  config_helper_.addConfigModifier([](envoy::config::bootstrap::v3::Bootstrap& bootstrap) {
+    auto* ads_config = bootstrap.mutable_dynamic_resources()->mutable_ads_config();
+    auto* grpc_service = ads_config->add_grpc_services();
+
+    std::string cluster_name = "ads_cluster";
+    grpc_service->mutable_envoy_grpc()->set_cluster_name(cluster_name);
+    auto* ads_cluster = bootstrap.mutable_static_resources()->add_clusters();
+    ads_cluster->MergeFrom(bootstrap.static_resources().clusters()[0]);
+    ads_cluster->set_name("ads_cluster");
+  });
+  setUpstreamProtocol(FakeHttpConnection::Type::HTTP2);
+  HttpIntegrationTest::initialize();
+  if (xds_stream_ == nullptr) {
+    createXdsConnection();
+    AssertionResult result = xds_connection_->waitForNewStream(*dispatcher_, xds_stream_);
+    RELEASE_ASSERT(result, result.message());
+    xds_stream_->startGrpcStream();
+  }
+}
+
+void XdsFuzzTest::close() {
+  cleanUpXdsConnection();
+  test_server_.reset();
+  fake_upstreams_.clear();
+}
+
+/**
+ * remove a listener from the list of listeners if it exists
+ * @param the listener number to be removed
+ * @return the listener as an optional so that it can be used in a delta request
+ */
+absl::optional<std::string> XdsFuzzTest::removeListener(uint32_t listener_num) {
+  std::string match = absl::StrCat("listener_", listener_num % ListenersMax);
+
+  for (auto it = listeners_.begin(); it != listeners_.end(); ++it) {
+    if (it->name() == match) {
+      listeners_.erase(it);
+      return match;
+    }
+  }
+  return {};
+}
+
+/**
+ * remove a route from the list of routes if it exists
+ * @param the route number to be removed
+ * @return the route as an optional so that it can be used in a delta request
+ */
+absl::optional<std::string> XdsFuzzTest::removeRoute(uint32_t route_num) {
+  std::string match = absl::StrCat("route_config_", route_num % RoutesMax);
+  for (auto it = routes_.begin(); it != routes_.end(); ++it) {
+    if (it->name() == match) {
+      routes_.erase(it);
+      return match;
+    }
+  }
+  return {};
+}
+
+/**
+ * wait for a specific ACK, ignoring any other ACKs that are made in the meantime
+ * @param the expected API type url of the ack
+ * @param the expected version number
+ * @return AssertionSuccess() if the ack was received, else an AssertionError()
+ */
+AssertionResult XdsFuzzTest::waitForAck(const std::string& expected_type_url,
+                                        const std::string& expected_version) {
+  if (sotw_or_delta_ == Grpc::SotwOrDelta::Sotw) {
+    API_NO_BOOST(envoy::api::v2::DiscoveryRequest) discovery_request;
+    do {
+      VERIFY_ASSERTION(xds_stream_->waitForGrpcMessage(*dispatcher_, discovery_request));
```

Comment:
Possibly what you are seeing is that the discovery request for RDS doesn't mention the ignored resource and so no match?